src/data/geo/polygon.js

```diff
@@ -0,0 +1,128 @@
+import Geometry from './geometry'
+import LineString from './linestring'
+import SpatialReferenceSystem from './spatial-reference-system'
+import { GeoTypes } from './constants'
+
+const GEOJSON_TYPE = GeoTypes.POLYGON
+const WKT_TYPE = GEOJSON_TYPE.toUpperCase()
+
+const DIFFERENT_SRS_ERROR_MSG = (
+  'Some of the \'lineStrings\' were created in the Spatial Reference System (SRS) ' +
+  'that differs from the current polygon \'srs\'.'
+)
+
+const INVALID_POINTS_NUMBER_ERROR_MSG = 'Invalid number of points in LineString, must me > 3'
+
+const NOT_CLOSED_LINESTRING_MSG = 'Some of the \'LineStrings\' aren\'t closed (first and last points must be equal).'
+
+const checkConsistence = (lineString, srs) => {
+  const points = lineString.getPoints()
+
+  if (lineString.getSRS() !== srs) {
+    throw new Error(DIFFERENT_SRS_ERROR_MSG)
+  }
+
+  if (points.length < 4) {
+    throw new Error(INVALID_POINTS_NUMBER_ERROR_MSG)
+  }
+
+  if (!points[0].equals(points[points.length - 1])) {
+    throw new Error(NOT_CLOSED_LINESTRING_MSG)
+  }
+}
+
+class Polygon extends Geometry {
+  constructor(boundary, holes, srs) {
+    srs = srs || SpatialReferenceSystem.DEFAULT
+    super(srs)
+
+    this.setBoundary(boundary)
+    this.setHoles(holes)
+  }
+
+  validate() {
```

Comment:
Where is the method used?

Reply:
Nowhere. It's for user purposes.

Reply:
yeah, but looks like this method will never throw any errors because we can't create an instance with wrong `boundary` or `holes` and after each set we validate the new values

Reply:
But we can mutate some of the points from polygon and thus this polygon may be invalidated.

Reply:
We decided to remove all validations and give this work to the database